src/services/stringCompletions.ts

```diff
@@ -431,7 +431,7 @@ function getStringLiteralCompletionEntries(sourceFile: SourceFile, node: StringL
             //      export * from "/*completion position*/";
             return { kind: StringLiteralCompletionKind.Paths, paths: getStringLiteralCompletionsFromModuleNames(sourceFile, node, compilerOptions, host, typeChecker, preferences) };
         case SyntaxKind.CaseClause:
-            const tracker = newCaseClauseTracker(typeChecker, (node.parent as CaseClause).parent.clauses);
+            const tracker = newCaseClauseTracker(typeChecker, (walkUpParenthesizedExpressions(node.parent) as CaseClause).parent.clauses);

```

Comment:
Instead of the double walk, you can just use `parent`, right?

Reply:
Ah, yes - i didn't recheck if the walked node is already available. And it is since it's actually what the switch cache here checks against.

```suggestion
            const tracker = newCaseClauseTracker(typeChecker, (parent as CaseClause).parent.clauses);

```